aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -554,6 +554,14 @@ module aptos_framework::coin {
         let maybe_supply = &mut borrow_global_mut<CoinInfo<CoinType>>(coin_address<CoinType>()).supply;
         if (option::is_some(maybe_supply)) {
             let supply = option::borrow_mut(maybe_supply);
+            spec {
+                use aptos_framework::optional_aggregator;
+                use aptos_framework::aggregator;
+                assume optional_aggregator::is_parallelizable(supply) ==> (aggregator::spec_aggregator_get_val(option::borrow(supply.aggregator))
+                    + amount <= aggregator::spec_get_limit(option::borrow(supply.aggregator)));
+                assume !optional_aggregator::is_parallelizable(supply) ==>
+                    (amount <= (option::borrow(supply.integer).limit - option::borrow(supply.integer).value));
```

Comment:
This may not be a problem, but could we use the form `a + b <= c` instead of `a <= c - b` just in case of underflow?

Reply:
Or, do we have a data invariant that `option::borrow(supply.integer).limit >= option::borrow(supply.integer).value)`

Reply:
yes, we have: https://github.com/aptos-labs/aptos-core/blob/22273a3ade7bcd3ccf4cabcff09dbd64c7aff0ea/aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.spec.move#L10

Reply:
Nice! Then, I think my comment earlier is not absolutely necessary.